runtime/lua/vim/snippet.lua

```diff
@@ -207,9 +213,60 @@ function Session.new(bufnr, snippet_extmark, tabstop_data)
     end
   end
 
+  self:set_keymaps()
+
   return self
 end
 
+--- Sets the snippet navigation keymaps.
+---
+--- @package
+function Session:set_keymaps()
+  local function maparg(key, mode)
+    local map = vim.fn.maparg(key, mode, false, true) --[[ @as table ]]
+    return not vim.tbl_isempty(map) and map or nil
+  end
+
+  local function set(jump_key, direction)
+    vim.keymap.set({ 'i', 's' }, jump_key, function()
+      return vim.snippet.active({ direction = direction })
+          and '<cmd>lua vim.snippet.jump(' .. direction .. ')<cr>'
+        or jump_key
```

Comment:
Is this even needed if we know we are already in an active snippet context? (Not that it hurts.)

Reply:
Yes it's needed because it's possible for the snippet to be active but not jumpable in the given direction (e.g. when modifying the last tabstop).